common/djangoapps/enrollment/views.py

```diff
@@ -668,6 +669,17 @@ def post(self, request):
             if email_opt_in is not None:
                 org = course_id.org
                 update_email_opt_in(request.user, org, email_opt_in)
+            else:
+                # This code is for REV-19 to handle the opt in flag
+                try:
+                    opt_in = ExperimentData.objects.get(
+                        user=request.user,
+                        experiment_id=9,
+                        key=unicode(course_id),
```

Comment:
suggestion: make the key something like "enroll_opt_in|course-v1:FooX+Bar+Baz". That way we can more clearly add other keys for this experiment.